src/node/snapshotter.h

```diff
@@ -248,16 +258,34 @@ namespace ccf
       std::lock_guard<std::mutex> guard(lock);
 
       CCF_ASSERT_FMT(
-        idx >= next_snapshot_indices.back(),
+        idx >= next_snapshot_indices.back().idx,
         "Committable seqno {} < next snapshot seqno {}",
         idx,
-        next_snapshot_indices.back());
+        next_snapshot_indices.back().idx);
+
+      bool forced = store->flag_enabled(
+        kv::AbstractStore::Flag::SNAPSHOT_AT_NEXT_SIGNATURE);
+
+      consensus::Index last_unforced_idx = last_snapshot_idx;
+      for (size_t i = next_snapshot_indices.size() - 1;
+           i < next_snapshot_indices.size();
+           i--)
```

Comment:
Could we make use of reverse iterators here instead of accessing `next_snapshot_indices` by index?

Reply:
Sure, np.